File: modules/weather.js
function weather(fullMessage){
	var phoneNumber = fullMessage.phoneNumber;
	var latitude = fullMessage.latitude.toString();
	var longitude = fullMessage.longitude.toString();
	var message = fullMessage.message;
	var weatherKey = '4e11214676001957120909';
	var forecastDays = '5'

	console.log(phoneNumber);
	console.log(latitude);
	console.log(longitude);
	console.log(message);

	var request = require('request');
	request('http://free.worldweatheronline.com/feed/weather.ashx?q='+latitude+','+longitude+'&format=json&num_of_days='+forecastDays+'&key='+weatherKey, function (error, response, body) {
	  if (!error && response.statusCode == 200) {
	    var jsonResponse = JSON.parse(body);
	    var currentTemp = jsonResponse.data.current_condition[0].temp_F + 'F';
	    var currentDesc = jsonResponse.data.current_condition[0].weatherDesc[0].value;
	    var moduleResponse = currentTemp + ' and ' + currentDesc;
	    console.log(moduleResponse);
	  }
	})
}

// Below is for testing

var fullMessage = {phoneNumber: '4153074175',
				   latitude: 37.775549,
				   longitude: -122.41378,
				   message: 'Whats the weather near blahblahblah?'};

weather(fullMessage);

File: modules.d/weather.py
#!/usr/bin/python
import sys
import re
import json
import urllib

# This is the input JSON that it sent to this module by the core program
input_jsonstr = ("\n".join(sys.stdin.readlines())).strip(" \n")
input = json.loads(input_jsonstr)

# Check that there is a subcommand
if (len(sys.argv) == 1):
    sys.stderr.write("Sub-command not given")
    sys.exit(1)

# This is the sub-command that this module is invoked with by the core program
subcommand = sys.argv[1]
if (subcommand == 'score'):
    if ("message" in input):
        match = re.search('weather', input['message'], flags=re.IGNORECASE);
        if (match):
            print 1
        else:
            print 0
    else:
        print 0
    sys.exit(0)

elif (subcommand == 'run'):
    latitude = str(input['latitude'])
    longitude = str(input['longitude'])
    numDays = '5'
    weatherKey = '4e11214676001957120909'
    response = urllib.urlopen('http://free.worldweatheronline.com/feed/weather.ashx?q='+latitude+','+longitude+'&format=json&num_of_days='+numDays+'&key='+weatherKey)
    for line in response:
        response_dict = json.loads(line)

    current_condition = response_dict['data']['current_condition'][0]
    current_temp = ['temp_F'] + 'F'
    weatherDesc = ['weatherDesc'][0]['value']

    print current_temp + ' and ' + weatherDesc
